src/appimagetool.c

```diff
@@ -131,13 +131,26 @@ int sfs_ls(char* image) {
 * execlp(), execvp(), and execvpe() search on the $PATH */
 int sfs_mksquashfs(char *source, char *destination, int offset) {
     pid_t pid = fork();
-    
     if (pid == -1) {
-        // error, failed to fork()
+        perror("sfs_mksquashfs fork() failed");
         return(-1);
-    } else if (pid > 0) {
+    }
+
+    if (pid > 0) {
+        // This is the parent process. Wait for the child to termiante and check its exit status.
         int status;
-        waitpid(pid, &status, 0);
+        if(waitpid(pid, &status, 0) == -1) {
+            perror("sfs_mksquashfs waitpid() failed");
+            return(-1);
+        }
+        
+        int retcode = WEXITSTATUS(status);
+        if (retcode) {
+            fprintf(stderr, "mksquashfs (pid %ld) exited with code %d\n", (long)pid, retcode);
```

Comment:
`pid_t` should be an `int` in glibc, so I'm not sure this cast is necessary, `%d` should work. (After all, `long int` and `int` are most likely the same size (4 bytes) on every viable platform anyway...)